File: crates/boundary/tests/fixtures/sample-java-project/src/main/java/com/example/application/UserService.java
package com.example.application;

import com.example.domain.user.User;
import com.example.domain.user.UserRepository;

public class UserService {
    private final UserRepository repo;

    public UserService(UserRepository repo) {
        this.repo = repo;
    }

    public User createUser(String name, String email) {
        User user = new User("generated-id", name, email);
        repo.save(user);
        return user;
    }
}

File: crates/boundary/tests/fixtures/sample-rust-project/src/application/user_service.rs
use crate::domain::user::{User, UserRepository};

pub struct UserService {
    repo: Box<dyn UserRepository>,
}

impl UserService {
    pub fn new(repo: Box<dyn UserRepository>) -> Self {
        Self { repo }
    }

    pub fn create_user(&self, name: &str, email: &str) -> Result<User, Error> {
        let user = User {
            id: "generated".to_string(),
            name: name.to_string(),
            email: email.to_string(),
        };
        self.repo.save(&user)?;
        Ok(user)
    }
}
